### ISSUE TITLE
Unable to locate iframe inside of a section

### ISSUE THREAD

**Issue by eRil** [cheezy/page-object#329]
Hi,
facing the issue:

```
  undefined method `switch_to' for #<Selenium::WebDriver::Element:0x007fdce3301e30> (NoMethodError)
```

Here's code:

``` ruby
class Section
  include PageObject

  in_iframe(:index => "1") do |frame|
    text_area(:email_recipients, id: 'recipients-area', :frame => frame)
  end
end

class MainPage
  include PageObject

  page_url("#{$app_home}")

  page_section(:status_gadget, Section, css: ".gadget-container")

end
```

When I trying to access element inside of iframe with

``` ruby
puts on_page(MainPage).status_gadget.email_recipients
```

I'm gettig error:

```
undefined method `switch_to' for #<Selenium::WebDriver::Element:0x007fdce3301e30> (NoMethodError)
```

But when I changing code to:

``` ruby
class MainPage
  include PageObject

  page_url("#{$app_home}")

  in_iframe(:index => "1") do |frame|
    text_area(:email_recipients, id: 'recipients-area', :frame => frame)
 end
```

And accessing:

``` ruby
puts on_page(MainPage).email_recipients
```

I'm getting expected output without errors.

**Comment by vveliev** [comment]
maybe you index is now "0" with respect to the section. Just an assumption

**Comment by vveliev** [comment]
I believe this is platform specific.

Selenium  - is state less.

if you really need iframe within section this may work for you ;) 

```rb
      def snippet_content(text)
        # in_iframe(:name => 'textarea_editor_ifr') do
        @browser.switch_to.frame('textarea_editor_ifr')
          @browser.find_element(:id, "tinymce").send_key(text)
        @browser.switch_to.default_content
        # end
      end

```

**Comment by AlexisKAndersen** [comment]
There is a fix for this going into the 2.2.0 release.
